### ISSUE TITLE
Class integration of columns, conditions and joins

### ISSUE THREAD

**Issue by mk3008** [mk3008/SqModel#138]
Parsing logic should be common.
Each builder process has its own uniqueness, so instead of complete class integration, consider whether it can be standardized by an interface or the like.

**Comment by mk3008** [comment]
Information gathering

### public class ValueExpression : IValueClause
general purpose
Can hold values, columns, boolean expressions.

### public class Condition : IQueryable, ICondition
has an ILogicalExpression

### public class ExistsExpression : ILogicalExpression
Have a SelectQuery.
It becomes an Exists statement.

### public class LogicalExpression : ILogicalExpression
has a left side and a right side.

### public class ConditionGroup : IQueryable, ICondition
Has a collection of IConditions.
Used for where clause, having clause.

### public interface IRelation : ICondition, IQueryable
have a left side table and a right side table
Although it is for the join expression, there is an omission because the join expression is not always written only by the table.

### public class RelationGroup : IQueryable, IRelation
have a left side table and a right side table.
It's better to think that you manage the join expression, but you manage the condition.
In other words, it is the same as the Where clause.
However, it is okay to have the concept of left side table and right side table, and it should be treated as a kind of group condition.

**Comment by mk3008** [comment]
ConditionGroup can manage left side table and right side table. #142

**Comment by mk3008** [comment]
Information gathering

### public class ValueExpression : IValueClause
general purpose
Can hold values, columns, boolean expressions.

### public class Condition : IQueryable, ICondition
has an ILogicalExpression

### public class ExistsExpression : ILogicalExpression
Have a SelectQuery.
It becomes an Exists statement.

### public class LogicalExpression : ILogicalExpression
has a left side and a right side.

### public class ConditionGroup : IQueryable, ICondition
Has a collection of IConditions.
Used for where clause, having clause.
